Rewrite "ProfilerStep#<num>" in profiler output

ilia-cher opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #51195 Add convenience import
* **#51194 Rewrite "ProfilerStep#<num>" in profiler output**
* #51192 Trim profiler file paths

Summary:
Aggregate all "ProfilerStep#<num>" together

Test Plan:
python test/test_profiler.py -k
test_kineto_profiler_api

Differential Revision: [D26113907](https://our.internmc.facebook.com/intern/diff/D26113907)

Commit:
Rewrite "ProfilerStep#<num>" in profiler output

Summary:
Aggregate all "ProfilerStep#<num>" together

Test Plan:
python test/test_profiler.py -k
test_kineto_profiler_api

[ghstack-poisoned]

ngimel commented on torch/autograd/profiler.py:
```diff
@@ -1101,6 +1102,14 @@ def filter_name(name):
     ]
     return name in filtered_out_names
 
+def rewrite_name(name, with_wildcard=False):
```
can you leave a comment what this function does? Demangles the name via StringTable?
Also, what is the purpose of wildcard argument?

Commit:
Update on "Rewrite "ProfilerStep#<num>" in profiler output"


Summary:
Aggregate all "ProfilerStep#<num>" together

Test Plan:
python test/test_profiler.py -k
test_kineto_profiler_api

Differential Revision: [D26113907](https://our.internmc.facebook.com/intern/diff/D26113907)

[ghstack-poisoned]
